src/lib/database/pocky-db.ts

```diff
@@ -41,6 +41,7 @@ export default class PockyDB implements PockyDbInterface {
 	 *         2 on error
 	 */
 	async givePegWithComment(comment : string, receiver : string, sender = 'default_user') : Promise<number> {
```

Comment:
This should let you give shame pegs even if all your pegs are spent.

ie. 
```
if (!senderHasPegs || !isShamePeg) {
 //block peg
}
//send peg
```

Reply:
You could refactor senderHasPegs to senderCanSendPeg instead, to cover this case.